diff --git a/crawl-ref/source/files.cc b/crawl-ref/source/files.cc
index f2d101a424..dbdcb01481 100644
--- a/crawl-ref/source/files.cc
+++ b/crawl-ref/source/files.cc
@@ -2625,6 +2625,11 @@ static void _save_game_exit()
     you.save = 0;
 }
 
+#ifdef __ANDROID__
+// Defined in libandroid.cc.
+extern "C" void android_notify_game_saved(const char *path);
+#endif
+
 void save_game(bool leave_game, const char *farewellmsg)
 {
     unwind_bool saving_game(crawl_state.saving_game, true);
@@ -2667,6 +2672,10 @@ void save_game(bool leave_game, const char *farewellmsg)
         {
             you.save->commit();
             save_game_prefs();
+#ifdef __ANDROID__
+            android_notify_game_saved(
+                get_savedir_filename(you.your_name).c_str());
+#endif
         }
         return;
     }
@@ -2675,6 +2684,10 @@ void save_game(bool leave_game, const char *farewellmsg)
     // so Valgrind doesn't complain.
     _save_game_exit();
 
+#ifdef __ANDROID__
+    android_notify_game_saved(get_savedir_filename(you.your_name).c_str());
+#endif
+
     game_ended(game_exit::save, farewellmsg ? farewellmsg
                                 : "See you soon, " + you.your_name + "!");
 }
