diff --git a/crawl-ref/source/known-items.cc b/crawl-ref/source/known-items.cc
index ccd19e2e06..3f0a820aef 100644
--- a/crawl-ref/source/known-items.cc
+++ b/crawl-ref/source/known-items.cc
@@ -126,56 +126,48 @@ protected:
         return Menu::process_key(key);
     }
 
-    string get_keyhelp(bool scrollable) const override
+    string get_keyhelp(bool) const override
     {
         string navigation;
         if (is_set(MF_NOSELECT))
         {
-            navigation = "<lightgrey>";
-            if (scrollable)
-            {
-                navigation +=
-                    menu_keyhelp_cmd(CMD_MENU_PAGE_DOWN) + " page down  "
-                    + menu_keyhelp_cmd(CMD_MENU_PAGE_UP) + " page up  ";
-            }
-            navigation += "[<w>-</w>] recognised  "
-                            + menu_keyhelp_cmd(CMD_MENU_EXIT) + " exit"
-                            "</lightgrey>";
+            navigation = "\n<lightgrey>"
+                "[<w>-</w>] recognised  "
+                            + menu_keyhelp_cmd(CMD_MENU_EXIT)
+                            + " exit</lightgrey>";
         }
         else
         {
-            // very similar to the MF_MULTISELECT case for regular Menus, but
-            // various differences require an override
-            navigation = "<lightgrey>" + menu_keyhelp_select_keys()
-                         + " select  ";
-
-            if (scrollable)
-            {
-                navigation +=
-                    menu_keyhelp_cmd(CMD_MENU_PAGE_DOWN) + " page down  "
-                    + menu_keyhelp_cmd(CMD_MENU_PAGE_UP) + " page up  ";
-            }
-            navigation += "</lightgrey>";
-            navigation = pad_more_with_esc(navigation);
-            navigation +=
-                    "\n<lightgrey>"
-                    "Letters toggle autopickup  ";
+            navigation = "\n<lightgrey>Letters toggle autopickup</lightgrey>";
+            navigation += "\n<lightgrey>"
+                + menu_keyhelp_select_keys()
+                + " select</lightgrey>";
+            navigation += "\n<lightgrey>"
+                + menu_keyhelp_cmd(CMD_MENU_PAGE_DOWN)
+                + " page down  "
+                + menu_keyhelp_cmd(CMD_MENU_PAGE_UP)
+                + " page up</lightgrey>";
             if (is_set(MF_ARROWS_SELECT))
             {
-                navigation += menu_keyhelp_cmd(CMD_MENU_TOGGLE_SELECTED)
-                    + " toggle selected  ";
+                navigation += "\n<lightgrey>"
+                    + menu_keyhelp_cmd(CMD_MENU_TOGGLE_SELECTED)
+                    + " toggle selected";
+                if (!all_items_known)
+                    navigation += "  [<w>-</w>] unrecognised";
+                navigation += "</lightgrey>";
             }
-
-            if (!all_items_known)
+            else if (!all_items_known)
             {
-                navigation +=
+                navigation += "\n<lightgrey>"
                     "[<w>-</w>] unrecognised"
                     "</lightgrey>";
             }
+            navigation += "\n<lightgrey>"
+                + menu_keyhelp_cmd(CMD_MENU_EXIT)
+                + " exit</lightgrey>";
         }
 
-        return pad_more_with(navigation,
-                            "<lightgrey>[<w>XXX</w>]</lightgrey>", MIN_COLS);
+        return navigation;
     }
 
     bool all_items_known;
