diff --git a/crawl-ref/source/known-items.cc b/crawl-ref/source/known-items.cc
index ccd19e2e06..43644f33c2 100644
--- a/crawl-ref/source/known-items.cc
+++ b/crawl-ref/source/known-items.cc
@@ -126,56 +126,52 @@ protected:
         return Menu::process_key(key);
     }
 
-    string get_keyhelp(bool scrollable) const override
+    bool process_command(command_type cmd) override
+    {
+        if (cmd == CMD_MENU_PAGE_DOWN || cmd == CMD_MENU_PAGE_UP)
+            return true;
+        return Menu::process_command(cmd);
+    }
+
+    string get_keyhelp(bool) const override
     {
         string navigation;
         if (is_set(MF_NOSELECT))
         {
-            navigation = "<lightgrey>";
-            if (scrollable)
-            {
-                navigation +=
-                    menu_keyhelp_cmd(CMD_MENU_PAGE_DOWN) + " page down  "
-                    + menu_keyhelp_cmd(CMD_MENU_PAGE_UP) + " page up  ";
-            }
-            navigation += "[<w>-</w>] recognised  "
-                            + menu_keyhelp_cmd(CMD_MENU_EXIT) + " exit"
-                            "</lightgrey>";
+            navigation = "<lightgrey>"
+                "[<w>-</w>] recognised  "
+                            + menu_keyhelp_cmd(CMD_MENU_EXIT)
+                            + " exit</lightgrey>";
         }
         else
         {
-            // very similar to the MF_MULTISELECT case for regular Menus, but
-            // various differences require an override
-            navigation = "<lightgrey>" + menu_keyhelp_select_keys()
-                         + " select  ";
-
-            if (scrollable)
-            {
-                navigation +=
-                    menu_keyhelp_cmd(CMD_MENU_PAGE_DOWN) + " page down  "
-                    + menu_keyhelp_cmd(CMD_MENU_PAGE_UP) + " page up  ";
-            }
-            navigation += "</lightgrey>";
-            navigation = pad_more_with_esc(navigation);
+            navigation = "<lightgrey>"
+                + menu_keyhelp_select_keys()
+                + " select</lightgrey>";
             navigation +=
-                    "\n<lightgrey>"
-                    "Letters toggle autopickup  ";
+                "\n<lightgrey>Letters toggle autopickup</lightgrey>";
             if (is_set(MF_ARROWS_SELECT))
             {
-                navigation += menu_keyhelp_cmd(CMD_MENU_TOGGLE_SELECTED)
-                    + " toggle selected  ";
+                navigation += "\n<lightgrey>"
+                    + menu_keyhelp_cmd(CMD_MENU_TOGGLE_SELECTED)
+                    + " toggle selected";
+                if (!all_items_known)
+                    navigation += "  [<w>-</w>] unrecognised";
+                navigation += "</lightgrey>";
             }
-
-            if (!all_items_known)
+            else if (!all_items_known)
             {
                 navigation +=
+                    "\n<lightgrey>"
                     "[<w>-</w>] unrecognised"
                     "</lightgrey>";
             }
+            navigation += "\n<lightgrey>"
+                + menu_keyhelp_cmd(CMD_MENU_EXIT)
+                + " exit</lightgrey>";
         }
 
-        return pad_more_with(navigation,
-                            "<lightgrey>[<w>XXX</w>]</lightgrey>", MIN_COLS);
+        return navigation;
     }
 
     bool all_items_known;
@@ -504,14 +500,14 @@ void check_item_knowledge(bool unknown_items)
     if (unknown_items)
         stitle = "Items not yet recognised:";
     else if (!all_items_known)
-        stitle = "Recognised items. (Select to toggle autopickup)";
+        stitle = "Recognised items.";
     else
-        stitle = "You recognise all items. (Select to toggle autopickup)";
+        stitle = "You recognise all items.";
 
-    string prompt = "(_ for help)";
-    //TODO: when the menu is opened, the text is not justified properly.
-    stitle = stitle + string(max(0, MIN_COLS - strwidth(stitle + prompt)),
-                             ' ') + prompt;
+    if (!unknown_items)
+        stitle += "\n(Select to toggle autopickup)  (_ for help)";
+    else
+        stitle += "\n(_ for help)";
 
     menu.set_preselect(&selected_items);
     menu_letter ml;
