diff --git a/crawl-ref/source/message.cc b/crawl-ref/source/message.cc
--- a/crawl-ref/source/message.cc
+++ b/crawl-ref/source/message.cc
@@ -538,7 +538,13 @@

     unsigned int out_width() const
     {
-        return width() - (use_first_col() ? 1 : 0);
+        unsigned int w = width() - (use_first_col() ? 1 : 0);
+        if (Options.msg_max_width > 0
+            && (unsigned int)Options.msg_max_width < w)
+        {
+            w = Options.msg_max_width;
+        }
+        return w;
     }

     unsigned int out_height() const
