diff --git a/crawl-ref/source/newgame.cc b/crawl-ref/source/newgame.cc
index 8ac5d80673..ec194e782c 100644
--- a/crawl-ref/source/newgame.cc
+++ b/crawl-ref/source/newgame.cc
@@ -1175,16 +1175,27 @@ public:

         welcome.textcolour(BROWN);
         welcome.cprintf("%s", _welcome(m_ng).c_str());
-        welcome.textcolour(YELLOW);
-        welcome.cprintf(" Please select your ");
-        welcome.cprintf(m_choice_type == C_JOB ? "background." : "species.");
-        m_vbox->add_child(make_shared<Text>(welcome));
+        if (m_choice_type == C_JOB)
+        {
+            m_vbox->add_child(make_shared<Text>(welcome));
+            formatted_string prompt;
+            prompt.textcolour(YELLOW);
+            prompt.cprintf("Please select your background.");
+            m_vbox->add_child(make_shared<Text>(prompt));
+        }
+        else
+        {
+            welcome.textcolour(YELLOW);
+            welcome.cprintf(" Please select your species.");
+            m_vbox->add_child(make_shared<Text>(welcome));
+        }

         descriptions = make_shared<Switcher>();

         m_main_items = make_shared<OuterMenu>(true, 3, 20);
         m_main_items->menu_id = m_choice_type == C_JOB ?
             "background-main" : "species-main";
+        m_main_items->linear_nav = true;
         m_main_items->set_margin_for_crt(1, 0);
         m_main_items->set_margin_for_sdl(15, 0);
         m_main_items->descriptions = descriptions;
@@ -1209,6 +1220,7 @@ public:
         m_sub_items = make_shared<OuterMenu>(false, 2, 4);
         m_sub_items->menu_id = m_choice_type == C_JOB ?
             "background-sub" : "species-sub";
+        m_sub_items->linear_nav = true;
         m_sub_items->descriptions = descriptions;
         m_vbox->add_child(m_sub_items);
         _add_choice_menu_options(m_choice_type, m_ng, m_defaults);
@@ -1858,6 +1870,7 @@ static bool _prompt_weapon(const newgame_def& ng, newgame_def& ng_choice,

     auto sub_items = make_shared<OuterMenu>(false, 2, 3);
     sub_items->menu_id = "weapon-sub";
+    sub_items->linear_nav = true;
     vbox->add_child(sub_items);

     main_items->linked_menus[2] = sub_items;
