diff --git a/crawl-ref/source/shopping.cc b/crawl-ref/source/shopping.cc
index b06d65869c..8ce9b7aa9d 100644
--- a/crawl-ref/source/shopping.cc
+++ b/crawl-ref/source/shopping.cc
@@ -1053,7 +1053,6 @@ ShopMenu::ShopMenu(shop_struct& _shop, const level_pos& _pos, bool _can_purchase
       can_purchase(_can_purchase)
 {
     menu_action = can_purchase ? ACT_EXECUTE : ACT_EXAMINE;
-    set_flags(get_flags() & ~MF_USE_TWO_COLUMNS);
 
     set_tag("shop");
 
@@ -1170,32 +1169,30 @@ void ShopMenu::update_help()
                                 && (!can_purchase ||
                                     no_selection && !from_shopping_list)
                 || menu_action == ACT_EXAMINE && !is_set(MF_ARROWS_SELECT);
-    const string action_key = no_action ? "       " // max: "[Enter]"
+    const string action_key = no_action ? ""
         : menu_keyhelp_cmd(CMD_MENU_ACCEPT_SELECTION);
     const string action_desc = action_key +
-            (no_action                   ? "                  "
-            : menu_action == ACT_EXAMINE ? " describe         "
+            (no_action                   ? ""
+            : menu_action == ACT_EXAMINE ? " describe"
             : from_shopping_list         ? " buy shopping list"
-                                         : " buy marked items ");
+                                         : " buy marked items");
 
     // XX swap shopping list / select by letter?
     const string mode_desc = !can_purchase ? ""
         : menu_keyhelp_cmd(CMD_MENU_CYCLE_MODE)
             + (menu_action == ACT_EXECUTE ? " <w>buy</w>|examine items"
                                           : " buy|<w>examine</w> items");
-    string m = menu_keyhelp_cmd(CMD_MENU_EXIT)+ " exit          ";
+    string m = menu_keyhelp_cmd(CMD_MENU_EXIT)+ " exit  ";
     m += mode_desc;
-    m = pad_more_with(m, make_stringf("%s %s",
-        hyphenated_hotkey_letters(item_count(), 'a').c_str(),
-        menu_action == ACT_EXECUTE ? "mark item for purchase   "
-                                   : "examine item             "));
-    m += make_stringf("\n[<w>/</w>] sort (%s)%s  %s",
+    if (!action_desc.empty())
+        m += "  " + action_desc;
+    m += make_stringf("\n[<w>/</w>] sort (%s)  %s %s",
         shopping_order_names[order],
-        string(7 - strwidth(shopping_order_names[order]), ' ').c_str(),
-        action_desc.c_str());
-
-    m = pad_more_with(m, hyphenated_hotkey_letters(item_count(), 'A')
-                                  + " put item on shopping list");
+        hyphenated_hotkey_letters(item_count(), 'a').c_str(),
+        menu_action == ACT_EXECUTE ? "mark item for purchase"
+                                   : "examine item");
+    m += "\n" + hyphenated_hotkey_letters(item_count(), 'A')
+                                  + " put item on shopping list";
 
 
     const string col = colour_to_str(channel_to_colour(MSGCH_PROMPT));
